willow/plugins/pillow.py

```diff
@@ -246,4 +254,144 @@ def to_buffer_rgba(self):
         return RGBAImageBuffer(image.size, image.tobytes())
 
 
-willow_image_classes = [PillowImage]
+class PillowAnimatedImage(Image):
+    def __init__(self, frames):
+        self.frames = frames
+
+    @classmethod
+    def check(cls):
+        _PIL_Image()
+
+    @classmethod
+    def is_format_supported(cls, image_format):
+        return is_format_supported(image_format)
+
+    @Image.operation
+    def get_size(self):
+        return self.frames[0].get_size()
+
+    @Image.operation
+    def get_frame_count(self):
+        return len(self.frames)
+
+    @Image.operation
+    def has_alpha(self):
+        return self.frames[0].has_alpha()
+
+    @Image.operation
+    def has_animation(self):
+        return self.get_frame_count() > 1
+
+    @Image.operation
+    def resize(self, size):
+        return PillowAnimatedImage([frame.resize(size) for frame in self.frames])
+
+    @Image.operation
+    def crop(self, rect):
+        return PillowAnimatedImage([frame.crop(rect) for frame in self.frames])
+
+    @Image.operation
+    def rotate(self, angle):
+        return PillowAnimatedImage([frame.rotate(angle) for frame in self.frames])
+
+    @Image.operation
+    def set_background_color_rgb(self, color):
+        return PillowAnimatedImage([frame.set_background_color_rgb(color) for frame in self.frames])
+
+    @Image.operation
+    def save_as_jpeg(self, f, quality=85, optimize=False, progressive=False):
+        if self.has_animation():
+            pass  # TODO: Raise warning
+
+        return self.frames[0].save_as_jpeg(f, quality=quality, optimize=optimize, progressive=progressive)
+
+    @Image.operation
+    def save_as_png(self, f, optimize=False):
+        if self.has_animation():
+            pass  # TODO: Raise warning
+
+        return self.frames[0].save_as_png(f, optimize=optimize)
+
+    @Image.operation
+    def save_as_gif(self, f):
+        image = self.frames[0].image
+        frames = self.frames
+
+        # All gif files use either the L or P mode but we sometimes convert them
+        # to RGB/RGBA to improve the quality of resizing. We must make sure that
+        # they are converted back before saving.
+        if image.mode not in ['L', 'P']:
+            frames = [
+                frame.convert('P', palette=_PIL_Image().ADAPTIVE)
+                for frame in frames
+            ]
+
+        params = {
+            'save_all': True,
+            'duration': image.info['duration'],
+            'append_images': [frame.image for frame in frames[1:]]
+        }
+
+        if 'transparency' in image.info:
+            params['transparency'] = image.info['transparency']
+
+        image.save(f, 'GIF', **params)
+
+        return GIFImageFile(f)
+
+    @Image.operation
+    def save_as_webp(self, f):
+        if self.has_animation():
+            pass  # TODO: Raise warning
+
+        return self.frames[0].save_as_png(f, optimize=optimize)
```

Comment:
Ooh, this is weird. It also shouldn't be calling `save_as_png`...

Reply:
.. but this doesn't exist anymore since `PillowAnimatedImage` has been merged into `PillowImage`

Reply:
That was on the old `save_as_png` method on `PillowAnimatedImage` - https://github.com/wagtail/Willow/pull/76/commits/f56e31746e50486e0ff8ec0e0a1e34a1e9bc90b5#diff-9bf9c7554d2d4846168b94a2dbc89951L309 - and is no longer relevant since it got merged into `PillowImage`